lib/sentry/envelope.ex

```diff
@@ -2,37 +2,35 @@ defmodule Sentry.Envelope do
   @moduledoc false
   # https://develop.sentry.dev/sdk/envelopes/
 
-  alias Sentry.{Config, Event, UUID}
+  alias Sentry.{Attachment, Config, Event, UUID}
 
   @type t() :: %__MODULE__{
           event_id: UUID.t(),
-          items: [Event.t(), ...]
+          items: [Event.t() | Attachment.t(), ...]
         }
 
   @enforce_keys [:event_id, :items]
   defstruct [:event_id, :items]
 
   @doc """
-  Creates a new envelope containing the given event.
-
-  Envelopes can only have a single element of type "event", so that's why we
-  restrict on a single-element list.
+  Creates a new envelope containing the given event and all of its attachments.
   """
-  @spec new([Event.t(), ...]) :: t()
-  def new([%Event{event_id: event_id}] = events) do
+  @spec from_event(Event.t()) :: t()
+  def from_event(%Event{event_id: event_id} = event) do
     %__MODULE__{
       event_id: event_id,
-      items: events
+      items: [event] ++ event.attachments
```

Comment:
maybe later let's add an `Envelope.add_item` method too when we need other types so that envelope is general enough

Reply:
Yep we'll probably need something like that down the line!